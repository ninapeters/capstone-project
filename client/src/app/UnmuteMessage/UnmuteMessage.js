import styled from 'styled-components'
import PropTypes from 'prop-types'
import { useState } from 'react'
import { motion } from 'framer-motion'
import { ArrowIcon } from '../Icons/Icons'

UnmuteMessage.propTypes = {
  children: PropTypes.string,
}

const variants = {
  open: { opacity: 1, y: 0 },
  closed: { opacity: 0, y: '-100%' },
}

export default function UnmuteMessage() {
  const [isShown, setIsShown] = useState(true)

  return (
    <MessageWrapper animate={isShown ? 'open' : 'closed'} variants={variants}>
      <p onClick={() => setIsShown(false)}>Please unmute your device.</p>
      <Arrow />
    </MessageWrapper>
  )
}

const MessageWrapper = styled(motion.section)`
  background: var(--white-transparent-min);
  border-radius: 20px;
  box-shadow: var(--shadow-light);
  display: flex;
  justify-content: center;
  padding: 8px;
  margin: 20px auto;
  width: 80%;
  p {
    color: var(--primary-regular);
    font-size: 0.7em;
    font-weight: 700;
    margin: 0;
    text-align: center;
  }
`
const Arrow = styled(ArrowIcon)`
  width: 10px;
  margin-left: 10px;
  fill: var(--primary-regular);
`
