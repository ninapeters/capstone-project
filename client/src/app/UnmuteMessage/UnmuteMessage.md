Unmute message:

```jsx
<UnmuteMessage onClick={() => {}} isShown={true} />
```
