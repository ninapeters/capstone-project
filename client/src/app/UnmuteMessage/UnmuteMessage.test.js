import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import UnmuteMessage from './UnmuteMessage'

describe('UnmuteMessage', () => {
  it('renders correctly', () => {
    const { container } = render(
      <UnmuteMessage onClick={() => {}} isShown={true} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
