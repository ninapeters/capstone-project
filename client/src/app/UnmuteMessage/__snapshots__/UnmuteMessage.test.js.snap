// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnmuteMessage renders correctly 1`] = `
.c0 {
  background: var(--white-transparent-min);
  border-radius: 20px;
  box-shadow: var(--shadow-light);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
  margin: 20px auto;
  width: 80%;
}

.c0 p {
  color: var(--primary-regular);
  font-size: 0.7em;
  font-weight: 700;
  margin: 0;
  text-align: center;
}

.c1 {
  width: 10px;
  margin-left: 10px;
  fill: var(--primary-regular);
}

<section
  class="c0"
  data-testid="message"
>
  <p>
    Please unmute your device.
  </p>
  <svg
    class="c1"
  >
    arrow.svg
  </svg>
</section>
`;
